tensorflow_probability/python/experimental/mcmc/particle_filter.py

```diff
@@ -670,6 +1122,8 @@ def _compute_observation_log_weights(step,
     observation = tf.nest.map_structure(
         lambda x, step=step: tf.gather(x, observation_idx), observations)
 
+    if particles_dim == 1:
+      observation = tf.expand_dims(observation, axis=0)
```

Comment:
I don't think this added code should be necessary.  The lines right below this should already handle any necessary `expand_dims` for `particles_dim`.

Reply:
I think it is necessary due to the dimensions mismatch

Reply:
With these two lines included, `ParticleFilterTestFloat32.test_batch_of_filters_particles_dim_1` fails when I run it.  If I delete these two lines, then that tests passes.